### ISSUE TITLE
features for UL re-miniAOD in 10_6_X

### ISSUE THREAD

**Issue by slava77** [cms-sw/cmssw#27889]
This is to keep track of features needed for the UL re-miniAOD.
The 2017 will likely happen first.
The target start date is still unclear (it will probably be around October 2019).

As of Aug 7  (after #31072 was merged) the re-miniAOD definitions aiming for UL re-miniAOD should use a  ```run2_miniAOD_UL```  modifier, if the configuration is to be controlled by a modifier. 

[as of Dec 9]
- [x] #27882 "Disabled PCA inputs for deepTau v2". Target: all re-miniAOD. (merged Sep 19, 2019)
    - the change to deepTau2017v2p1 is already applied in UL 2016 and 2018 by default
    - controlled by run2_miniAOD_devel
- [x] ~#28845 "Add customizer for PUPPI tune v13" (merged Feb 27, 2020)~
    - this is implemented with a customize method; superseded by v14
- [x] #29601 PUPPI tune v14 (merged May 14, 2020)
- [x] #29045 Adding parameters needed for PUjetID 94X and 102X (merged Mar 3, 2020)
    - bugfix for run2_miniAOD_devel -based setup in #29377 (merged April 2, 2020)
    - controlled by run2_miniAOD_devel
- [x] #29330 Fix MET unclustered energy computation (merged April 1, 2020)
    - controlled by run2_miniAOD_devel
- [x] #29392 Fix MET Significance, Covariance Matrix and unclustered energy lepton subtraction (merged Apr 21, 2020)
    - controlled by run2_miniAOD_devel
- [x] #29521 Keeping sumPt from MET Significance calculation  (merged Apr 21, 2020)
    - enabled, no era modifiers
- [x] #29062 MXNet-based implementation of the ParticleNet tagger  (merged Apr 18, 2020)
    - controlled by run2_miniAOD_devel
- [x] #29324 inverse beta variable in pat::Muons (merged Apr 17, 2020)
    - enabled, no era modifiers
- [x] #29529 Update of the lostTrack collection for MUO  (merged Apr 23, 2020)
- [x] #29537 fix sign of pat::Muon::dxy (merged May 1, 2020)
- [x] #29771 PPS rereco-in-miniAOD (merged May 25, 2020)
    - controlled by run2_miniAOD_devel
- [x] #29739 Tau againstElectronDeadECAL (merged May 27, 2020 )
    - controlled by run2_miniAOD_devel
- [x] #30066 Fix bug affecting boosted tau isolation (merged June 4, 2020)
- [x] #30199 Updates to electron/photon scales and smearings for UL (merged June 19, 2020)
- [x] cms-sw/cmsdist#5902 update ONNXRuntime to 1.3.0 (merged June 17, 2020)
- [x] #30123 ONNXRuntime-based implementation of DeepJet, DeepAK8 and DeepDoubleX (technical; significant speedup) (merged July 2, 2020)
- [x] #30352 define UL re-miniAOD workflows; technical at this point, repeats UL main miniAOD (merged June 26, 2020)
- [x] #30542 deepDoubleX V2; capability only, plan to enable later (merged July 13, 2020)
- [x] #30584 Hbb Interaction network tagger (merged July 20, 2020)
- [x] #30892 Use dxy sign of GSF track instead of IPTools for pat::Electron (merged Aug 4, 2020)
- [x] #30935 Pile-Up ID training and WPs for 2017 UL samples (merged Aug 4, 2020)
- [x] #31072 migrate run2_miniAOD_devel parts suitable for UL re-mini to use run2_miniAOD_UL (merged Aug 7, 2020)
- [x] #31044  fix to enable hpsPFTauDiscriminationByDeadECALElectronRejection in run2 reminiAOD (merged Aug 16, 2020)
- [x] #31037 updated training for ParticleNet tagger (merged Aug 17, 2020)
- [x] #31214 Adapt track-vertex association in PUPPI (v15) (merged Aug 25, 2020)
- [x] #31120 deepMet for nanoAOD, includes an operational bugfix in DeepMETProducer (merged Aug 25, 2020)
- [x] #31087 bad PF muon filter (merged Sep 14, 2020)
- [x] #31065  Update tau anti electron discriminators (merged Sep 1, 2020)
- [x] #31290 improving (computational) performance of PuppiProducer (merged Sep 4, 2020)
- [x] #31413 HF Noise Event Filter (merged Sep 16, 2020)
- [x] #31420 HF jet shower shape (merged Sep 18, 2020)
- [x] #31510 HCAL slimmed collections (merged Sep 23, 2020)
- [x] #31615 ParticleNet AK4 jet tagger (merged Oct 7, 2020)
- [x] #31847 Activate deepDoubleXV2 (merged Nov 4, 2020)
- [x] #31995 Electron Scale and smearing for UL (merged Nov 4, 2020)
- [x] #32055 speedup at job start in EnergyScaleCorrection (merged Nov 10, 2020)
- [x] #32303 Expand track information in lostTracks collection, for tau reco-from-mini (merged Nov 30, 2020)
- [x] #31984 "unprefireable bit info" (merged Dec 2, 2020)
- [x] #32372 update LowPtElectrons in bParking config and add low-pt electrons in default miniAOD (merged Dec 9, 2020)

The following features were descoped:
- (no 10_6_X PR) #32391 energy regression for low-pt electrons, including updated training of IDs (all can be redone from miniAOD, not in #32391 as of Dec 8)
- quark gluon likelihood retraining (controlled by GT; possibly descoped https://github.com/cms-sw/cmssw/issues/27889#issuecomment-726827706)
- update of the prefiring maps. (will update a file in https://github.com/cms-data/PhysicsTools-PatUtils; possibly descoped https://github.com/cms-sw/cmssw/issues/27889#issuecomment-726827706)

A bit outside of re-miniAOD scope (requires fresh MC/rereco)
- [x] #31350 Store TrackExtra, TrackingRecHits, and SiPixel/StripClusters for muons in AOD+MINIAOD

**Comment by slava77** [comment]
it sounds like there is still time for deploying the fix in the 2018 UL.
I propose to make ```run2_tau_ul_2018``` and ```run2_tau_ul_2016``` modifiers which can be added to ```Run2_2018``` and ```Run2_2016``` eras.
There is not need to propagate these to the master.

@fabiocos @kpedro88 @srimanob this is as a follow up to the discussion today in the ORP.
Please check if this proposal makes sense to you as well.

**Comment by peruzzim** [comment]
> it sounds like there is still time for deploying the fix in the 2018 UL.
> I propose to make `run2_tau_ul_2018` and `run2_tau_ul_2016` modifiers which can be added to `Run2_2018` and `Run2_2016` eras.
> There is not need to propagate these to the master.
> 
> @fabiocos @kpedro88 @srimanob this is as a follow up to the discussion today in the ORP.
> Please check if this proposal makes sense to you as well.

I agree with the above, but please let's keep it active also under the generic devel modifier, so we will remember to add it also to 2017 in a future re-miniAOD of that.

**Comment by peruzzim** [comment]
taking note of https://github.com/cms-sw/cmssw/pull/27929#issuecomment-547892390

**Comment by peruzzim** [comment]
taking note of #28844

**Comment by peruzzim** [comment]
taking note of #25884

**Comment by peruzzim** [comment]
also taking note of missing unprefireable bit information in UL miniAOD

The envisaged solution is:
- run something equivalent to the `L1Repack:uGT` step, re-emulating `ExternalAlgBlock` in data with a dedicated customisation
- adapt NanoAOD configuration accordingly

@rekovic @srimanob

**Comment by peruzzim** [comment]
assign l1

**Comment by srimanob** [comment]
FYI @ameyer-cern

**Comment by peruzzim** [comment]
taking note of MUO request for additional tracks to be added in MiniAOD supporting SF derivation without access to AOD

@dntaylor

**Comment by slava77** [comment]
@danbarto @ahinzmann 
I suppose that we also need a backport for #28370

**Comment by ahinzmann** [comment]
Will need also #29392

**Comment by peruzzim** [comment]
taking note of #29062

**Comment by slava77** [comment]
I've updated the issue description https://github.com/cms-sw/cmssw/issues/27889#issue-486344664 with the list of features mentioned in the thread.
Please check if something is missing

**Comment by slava77** [comment]
I missed the PPS plans; added now in the issue description

@jan-kaspar @fabferro

**Comment by jan-kaspar** [comment]
> I missed the PPS plans; added now in the issue description
> 
> @jan-kaspar @fabferro

Thanks @slava77 . I confirm that we use run2_miniAOD_devel to activate the changes for re-miniAOD, e.g.:
  https://github.com/CTPPS/cmssw/blob/jan_buffer_10_6/RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py#L32

**Comment by slava77** [comment]
@jainshilpi @lsoffi @afiqaize @SohamBhattacharya 
Please clarify if there is something planned from EGM (apparently something was mentioned in private)

**Comment by hqucms** [comment]
Would it make sense to backport https://github.com/cms-sw/cmssw/pull/28112 to benefit from the speed improvement?

**Comment by slava77** [comment]
> Would it make sense to backport #28112 to benefit from the speed improvement?

I don't have a strong feeling that it has to be done.
A backport would likely need some form of #29172 (to have ONNX only for re-mini and allow the older workflows to run what they do now).

**Comment by slava77** [comment]
I updated the issue description (to keep a summary in one place)
- PUPPI tune v14 is mentioned 
- updated the line item for sumPtUnclustered, now that the PR is available #29521 
- updated status after #29062 and #29324 were merged

**Comment by slava77** [comment]
Update from today:
- #29392 is merged
- #29521 is merged
- lowPt electrons work (details in #25884) has an ETA of ~1 month
- added #29529 (step up from "backport needed" for #29389 )

**Comment by slava77** [comment]
I added #29537 to fix sign of pat::Muon::dxy in the list of features in the description/first message in the issue

**Comment by slava77** [comment]
updates from today
- PUPPI tune v14 now has a PR #29601 
- #29529 lostTrack was merged

**Comment by slava77** [comment]
also, added a note on another feature
- (backport needed for) #29526 Updates to electron/photon scales and smearings for UL

**Comment by mbluj** [comment]
There is intention to add tauID against electrons in deadECal regions (https://github.com/cms-sw/cmssw/pull/29739) to future UL reMiniAOD productions.

**Comment by slava77** [comment]
updates from today:
- #29601 PUPPI v14 not on status (not merged yet)
- #29537 Muon dxy : merged May 1
- #29771 PPS remini (a feature now with a number) (not merged yet)
- #29739 Tau againstElectronDeadECAL (not merged yet)
- #29756 (+ part of #29826) run2_miniAOD_UL , to be backported

**Comment by slava77** [comment]
update:
- #29601 PUPPI v14 not on status : merged May 14

**Comment by slava77** [comment]
updates:
- added new feature (backport of ) cms-sw/cmsdist#5857 update ONNXRuntime to 1.3.0
- #29771 PPS rereco-in-miniAOD (merged May 25, 2020)
- #29739 Tau againstElectronDeadECAL (signed by reco May 26, 2020 )

**Comment by slava77** [comment]
updates:
- #29739 Tau againstElectronDeadECAL (merged May 27, 2020)
- features not yet merged are now at the bottom of the list

**Comment by slava77** [comment]
update
- #30066 "Fix bug affecting boosted tau isolation" added to the list of features/fixes

**Comment by slava77** [comment]
updates (in the issue description)
- #30199 added (backport of #29526 is now available) for e/gamma scales and smearing
- #30066 Fix bug affecting boosted tau isolation (merged June 4, 2020)
- (backport is now available) cms-sw/cmsdist#5926 update ONNXRuntime to 1.3.0
- #30123 ONNXRuntime-based implementation of DeepJet, DeepAK8 and DeepDoubleX

**Comment by slava77** [comment]
updates, including after the reco meeting today:
- (backport needed for) #30015 bad PF muon filter
- (possible backport for) #30016 deepDoubleX V2
- (possible backport for) #30072 Hbb Interaction network tagger (could be nanoAOD only)

The taggers can run on top of miniAOD (in case the integration takes some time)

**Comment by slava77** [comment]
updates (in the issue description
- cms-sw/cmsdist#5902 replaced cms-sw/cmsdist#5926

**Comment by slava77** [comment]
updates:
- #30199 Updates to electron/photon scales and smearings for UL (merged June 19, 2020)
- cms-sw/cmsdist#5902 update ONNXRuntime to 1.3.0 (merged June 17, 2020)
- #30123 ONNXRuntime-based implementation of DeepJet, DeepAK8 and DeepDoubleX (technical; significant speedup) (merged July 2, 2020)
- #30542 deepDoubleX V2; capability only, plan to enable later (backport PR is now available)
- #30584 Hbb Interaction network tagger (backport PR is now available)
- #30352 define UL re-miniAOD workflows; technical at this point, repeats UL main miniAOD (merged June 26, 2020)
    - updated description/context for the follow up: migrate run2_miniAOD_devel parts suitable for UL re-mini to use run2_miniAOD_UL

**Comment by slava77** [comment]
updates:
- #30542 deepDoubleX V2; capability only, plan to enable later (merged July 13, 2020)
- #30584 Hbb Interaction network tagger (merged July 20, 2020)

**Comment by slava77** [comment]
update
- (backport needed for) #30738 Use dxy sign of GSF track instead of IPTools for pat::Electron

**Comment by slava77** [comment]
> update
> 
>     * (backport needed for) #30738 Use dxy sign of GSF track instead of IPTools for pat::Electron

no updated with a 10_6_X PR info #30892

**Comment by slava77** [comment]
a couple of tau-related notes:
- need a resolution for #31015 
- double-check if a backport of #27465 is not needed

**Comment by slava77** [comment]
> a couple of tau-related notes:
> 
>     * need a resolution for #31015

provided in #31044 now

> 
>     * double-check if a backport of #27465 is not needed

a backport will be made according to https://github.com/cms-sw/cmssw/pull/27465#issuecomment-669294131

**Comment by mbluj** [comment]
> 
> 
> > a couple of tau-related notes:
> > ```
> > * need a resolution for #31015
> > ```
> 
> provided in #31044 now
> 
> > ```
> > * double-check if a backport of #27465 is not needed
> > ```
> 
> a backport will be made according to [#27465 (comment)](https://github.com/cms-sw/cmssw/pull/27465#issuecomment-669294131)

provided in #31065 now (also includes changes from #31044)

**Comment by slava77** [comment]
Updated notes:
- #30892 Use dxy sign of GSF track instead of IPTools for pat::Electron is now merged, as of  Aug 4, 2020
- #30935 Pile-Up ID training and WPs for 2017 UL samples was merged Aug 4, 2020
- the following now have a PR number
    - #31044  fix to enable hpsPFTauDiscriminationByDeadECALElectronRejection in run2 reminiAOD
    - #31065  Update tau anti electron discriminators
    - #31072 migrate run2_miniAOD_devel parts suitable for UL re-mini to use run2_miniAOD_UL 
- New feature/update
    - #31037 updated training for ParticleNet tagger

**Comment by slava77** [comment]
one more PR backport is now known: #31087 bad PF muon filter

**Comment by slava77** [comment]
Updates:
- merged:
    - #31072 migrate run2_miniAOD_devel parts suitable for UL re-mini to use run2_miniAOD_UL (merged Aug 7, 2020)
    - #31044  fix to enable hpsPFTauDiscriminationByDeadECALElectronRejection in run2 reminiAOD (merged Aug 16, 2020)
    - #31037 updated training for ParticleNet tagger (merged Aug 17, 2020)
- added
    - #31120 deepMet for nanoAOD, includes an operational bugfix in DeepMETProducer
    - (backport needed for) #31174 Adapt track-vertex association in PUPPI (v15)
    - (potential to backport) #31164 improving (computational) performance of PuppiProducer

Also, a note about the use of the process modifier was updated to `
As of Aug 7  (after #31072 was merged) the re-miniAOD definitions aiming for UL re-miniAOD should use a  ```run2_miniAOD_UL```  modifier, if the configuration is to be controlled by a modifier. `

**Comment by slava77** [comment]
Update
- (potential backport for) #31212 bug-fix in metNoHF unclustered energy uncertainty

**Comment by slava77** [comment]
updates:
- removed "(potential backport for) #31212 bug-fix in metNoHF unclustered energy uncertainty " based on https://github.com/cms-sw/cmssw/pull/31212#issuecomment-681972929
- merged PRs area now checked as done:
    - #31214 Adapt track-vertex association in PUPPI (v15) (merged Aug 25, 2020)
    - #31120 deepMet for nanoAOD, includes an operational bugfix in DeepMETProducer (merged Aug 25, 2020)

**Comment by slava77** [comment]
updates from Sep 17

- resolved/done
    - #31087  bad PF muon filter (merged Sep 14, 2020)
    - #31065  Update tau anti electron discriminators (merged Sep 1, 2020)
    - #31290 improving (computational) performance of PuppiProducer (merged Sep 4, 2020)
    - #31413 HF Noise Event Filter (merged Sep 16, 2020)
- new 
    - (not really new, but added explicitly only now) add "Activate deepDoubleXV2"
    - #31420 HF jet shower shape
    - (A bit outside of re-miniAOD scope (requires fresh MC/rereco)) #31350 Store TrackExtra, TrackingRecHits, and SiPixel/StripClusters for muons in AOD+MINIAOD

**Comment by lathomas** [comment]
Hello. 
Two more things we hope to have for UL reminiaod on the JME side:  
- quark gluon likelihood retraining
- update of the prefiring maps. (this one would just be an update of the related file in https://github.com/cms-data/PhysicsTools-PatUtils in principle)

**Comment by slava77** [comment]
> Two more things we hope to have for UL reminiaod on the JME side:
> 
>     * quark gluon likelihood retraining
> 
>     * update of the prefiring maps. (this one would just be an update of the related file in https://github.com/cms-data/PhysicsTools-PatUtils in principle)

I added these to the feature list.

A couple of items were updated in the list as well:
- #31420 HF jet shower shape is now merged (Sep 18, 2020)
- "#31510 HCAL slimmed collections" is added to the list of features

**Comment by slava77** [comment]
updates from Oct 2:
-  #31510 HCAL slimmed collections  was merged Sep 23, 2020)
- new feature added: #31615 ParticleNet AK4 jet tagger
- info updated with master PR now available ( #31618) for Activate deepDoubleXV2

**Comment by slava77** [comment]
> updates from Oct 2:
> 
>     * #31510 HCAL slimmed collections  was merged Sep 23, 2020)
> 
>     * new feature added: #31615 ParticleNet AK4 jet tagger
> 
>     * info updated with master PR now available ( #31618) for Activate deepDoubleXV2

also, added a note for "produce lowPtGsfEle slimmed collections": Target by October 10 or so.

**Comment by slava77** [comment]
updates
- #31615 ParticleNet AK4 jet tagger (merged Oct 7, 2020)

added
-  (backport needed for) #31220 update LowPtElectrons in bParking config
    - this is a part of resolution to #25884

**Comment by mariadalfonso** [comment]
update
* Electron Scale and smearing for UL 
(backport needed for) https://github.com/cms-sw/cmssw/pull/31936

**Comment by slava77** [comment]
> update
> 
>     * Electron Scale and smearing for UL
>       (backport needed for) #31936

this is added now in the description

also updated at the top:
- #31847 Activate deepDoubleXV2 (now a 10_6_X PR)
- #31984 "unprefireable bit info"
    - only the feature is in, customization `L1TStage2SetPrefireVetoBit` needs to be propagated to the workflows
- #31220 update LowPtElectrons ... : the item was merged with  #25884 now that #31220 includes the low-pt elements for the default miniAOD

**Comment by slava77** [comment]
updates:
- now merged
    - #31847 Activate deepDoubleXV2 (merged Nov 4, 2020)
    - #31995 Electron Scale and smearing for UL (merged Nov 4, 2020)
    - #32055 speedup at job start in EnergyScaleCorrection (merged Nov 10, 2020)

**Comment by slava77** [comment]
@lathomas @ahinzmann @kirschen @cms-sw/xpog-l2 
please comment on the status of the following items which do not yet have a PR : 
- quark gluon likelihood retraining
- update of the prefiring maps. (this one would just be an update of the related file in https://github.com/cms-data/PhysicsTools-PatUtils in principle)

**Comment by lathomas** [comment]
Both are ongoing. 2017 prefiring maps could be ready soon. 2016 will take longer (need UL16 JECs as inputs). 
For qgl, it might take some time. This would be nice to have in reMINIAOD but not critical (i.e. I wouldn't delay everything for this).

**Comment by slava77** [comment]
> Both are ongoing. 2017 prefiring maps could be ready soon. 2016 will take longer (need UL16 JECs as inputs).
> For qgl, it might take some time. This would be nice to have in reMINIAOD but not critical (i.e. I wouldn't delay everything for this).

Thank you for clarifying.
Do I read your comment correctly that the prefiring maps are still a required feature for remini,
while the qgl is more of a nice to have?

**Comment by lathomas** [comment]
I don't think that the prefiring maps are such a big deal than they justify delaying reMINIAOD, given that the prefiring weights can be recomputed on top of MINIAOD. 

That being said, the timeline is the following: 
2017 maps should hopefully come soon (in a week or two). 
2016 maps won't be ready before JECs so it will probably take a month (but maybe reMINI of UL16 MC won't have started yet?).

By the way, the prefiring weights are NOT saved in MINIAOD right now. The producer is however used to compute the weights that are stored NANOAOD, and this is why we would care (but then a future reNANO could add them)

**Comment by slava77** [comment]
@bainbrid 
please list the remaining features for low-pt electrons that will not enter #31220 and are planned for separate PRs

**Comment by bainbrid** [comment]
In response to https://github.com/cms-sw/cmssw/issues/27889#issuecomment-726867673:

For low-pT electrons:
- we confirm that we will add code/configuration to perform an energy regression using RECO/AOD inputs
- we are considering to embed flags that identify if the pat::Electron originates from a converted photon
- we are considering to allow updates to the regression and ID using MINIAOD inputs ("MiniAODfromMiniAOD")

@crovelli @fcavallo @nancymarinelli

**Comment by perrotta** [comment]
Additional tracks information for the lostTracks, which was submitted by the tau group in the master with #32242, could be considered for the backport, if it can be done in the shadow of the other pending pull requests (see https://github.com/cms-sw/cmssw/pull/32242#issuecomment-732260157)

**Comment by perrotta** [comment]
> In response to [#27889 (comment)](https://github.com/cms-sw/cmssw/issues/27889#issuecomment-726867673):
> 
> For low-pT electrons:
> 
> * we confirm that we will add code/configuration to perform an energy regression using RECO/AOD inputs
> * we confirm the need for a final ID training (to replace the current placeholder) and associated PR for 11_2 and 10_6.
> * we are considering to embed flags that identify if the pat::Electron originates from a converted photon
> * we are considering to allow updates to the regression and ID using MINIAOD inputs ("MiniAODfromMiniAOD")
> 
> @crovelli @fcavallo @nancymarinelli

@bainbrid I imagine you were supposed to tag Francesca Cavallari (of whom I don't know the github name, sorry...) instead of Francesca Cavallo (Muon DPG)

**Comment by bainbrid** [comment]
> > @crovelli @fcavallo @nancymarinelli
> 
> @bainbrid I imagine you were supposed to tag Francesca Cavallari (of whom I don't know the github name, sorry...) instead of Francesca Cavallo (Muon DPG)

Fixed, thanks! @cavalari

**Comment by slava77** [comment]
updates from today (now in the issue description/list):
- added
    -  #32303 Expand track information in lostTracks collection, for tau reco-from-mini
    -  (no PR/PRs) energy regression for low-pt electrons, including updated training of IDs (all can be redone from miniAOD, can be descoped)
- status/description updated
    -  #31984 "unprefireable bit info" (fully signed Nov 26, 2020)
    -  (backport needed for) #31220 update LowPtElectrons in bParking config and add low-pt electrons in default miniAOD (signed by reco Nov 26, 2020)
    -  quark gluon likelihood retraining (controlled by GT; possibly descoped https://github.com/cms-sw/cmssw/issues/27889#issuecomment-726827706)
    -  update of the prefiring maps. (will update a file in https://github.com/cms-data/PhysicsTools-PatUtils; possibly descoped https://github.com/cms-sw/cmssw/issues/27889#issuecomment-726827706)

**Comment by slava77** [comment]
updates from near the finish line :
- merged:
    - #32303 Expand track information in lostTracks collection, for tau reco-from-mini (merged Nov 30, 2020)
    - #31984 "unprefireable bit info" (merged Dec 2, 2020)
- pending, but almost ready
    - #32372 update LowPtElectrons in bParking config and add low-pt electrons in default miniAOD (signed by reco Dec 8, 2020)
- remaining features were descoped (all expected to be re-doable from miniAOD inputs, which makes these candidates for the next nanoAOD)
    - low-pt electron energy regression (controlled by GT, but also needs code updates) and 2020Nov28 ID 
    - quark gluon likelihood retraining (controlled by GT)
    - update of the prefiring maps

**Comment by slava77** [comment]
+1

it sounds like we are in a production stage now

**Comment by mariadalfonso** [comment]
+xpog

indeed production advancing

**Comment by srimanob** [comment]
@cms-sw/operations-l2 should this issue be closed?

**Comment by perrotta** [comment]
> @cms-sw/operations-l2 should this issue be closed?

I think so
